core/commands/version.go

```diff
@@ -83,3 +88,50 @@ var VersionCmd = &cmds.Command{
 	},
 	Type: VersionOutput{},
 }
+
+type Dependency struct {
+	Path       string
+	Version    string
+	ReplacedBy string
+	Sum        string
+}
+
+var depsVersionCommand = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Shows information about dependencies used for build",
+		ShortDescription: `
+Print out all dependencies and their versions.`,
+	},
+	Type: Dependency{},
+
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		info, ok := debug.ReadBuildInfo()
+		if !ok {
+			return errors.New("no embedded dependency information")
+		}
+		toDependency := func(mod *debug.Module) (dep Dependency) {
+			dep.Path = mod.Path
+			dep.Version = mod.Version
+			dep.Sum = mod.Sum
+			if repl := mod.Replace; repl != nil {
+				dep.ReplacedBy = fmt.Sprintf("%s@%s", repl.Path, repl.Version)
+			}
+			return
+		}
+		res.Emit(toDependency(&info.Main))
+		for _, dep := range info.Deps {
+			res.Emit(toDependency(dep))
+		}
+		return nil
+	},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, dep Dependency) error {
+			fmt.Fprintf(w, "%s@%s", dep.Path, dep.Version)
+			if dep.ReplacedBy != "" {
+				fmt.Fprintf(w, " => %s", dep.ReplacedBy)
+			}
+			fmt.Fprintf(w, "\n")
+			return errors.New("test")
```

Comment:
foobar

Reply:
Seems like this error gets ignored. Interesting.

Reply:
```
diff --git a/core/commands/version.go b/core/commands/version.go
index 9c5c666d0..c9632f799 100644
--- a/core/commands/version.go
+++ b/core/commands/version.go
@@ -118,9 +118,13 @@ Print out all dependencies and their versions.`,
 			}
 			return
 		}
-		res.Emit(toDependency(&info.Main))
+		if err := res.Emit(toDependency(&info.Main)); err != nil {
+			return err
+		}
 		for _, dep := range info.Deps {
-			res.Emit(toDependency(dep))
+			if err := res.Emit(toDependency(dep)); err != nil {
+				return err
+			}
 		}
 		return nil
 	},
```